### PR TITLE
Implementation for H1 and makefile

### PR DESCRIPTION
**Pull request by gertjaap**
Implemented H1

Using `index = 16` it generates a file of roughly 1GB (`1,140,850,016` bytes).

Using `sha3("Vertcoin PoS PoC")` as input to generating the file.

Also added a makefile and README

### CODE REVIEW

File: h1.c [hunk 1]
```diff
@@ -132,11 +138,299 @@ void ButterflyGraph(struct Graph* g, int64_t index, int64_t* count) {
...
+                NewNode(g, count, hashOutput);
+                count++;
+            }
+        } else {
+            uint64_t sinks = count;
+            uint64_t sources = sinks + pow2indexInner - numXi(index);
+            for(i = 0; i < pow2indexInner_1; i++) {
+                uint64_t nodeId0 = sinks + i;
+				uint64_t nodeId1 = sinks + i + pow2indexInner_1;
+				uint8_t* parent0 = GetNode(g, sinks - pow2indexInner_1 + i);
```
Comment (metalicjames) [inline h1.c]: Not sure if this is a display bug, but should fix the strange indentation.

File: h1.c [hunk 2]
```diff
@@ -132,11 +138,299 @@ void ButterflyGraph(struct Graph* g, int64_t index, int64_t* count) {
...
+				uint8_t* parent1_1 = GetNode(g, sources + i + pow2indexInner_1);
+
+                uint8_t* buf = malloc(NODE_SIZE);
+                WriteVarInt(buf, nodeId0);
+                uint8_t* hashInput = malloc(NODE_SIZE * 4);
+                memcpy(hashInput, g->pk, NODE_SIZE);
+                memcpy(hashInput+NODE_SIZE, buf, NODE_SIZE);
+                memcpy(hashInput+(NODE_SIZE*2), parent0, NODE_SIZE);
+                memcpy(hashInput+(NODE_SIZE*3), parent1_0, NODE_SIZE);
+                //printf("Hash input 0: ");
```
Comment (metalicjames) [inline h1.c]: Remove the commented out code.

File: h1.c (hunk 1 above)
Comment (gertjaap) [reply h1.c]: Fixed

File: h1.c (hunk 2 above)
Comment (gertjaap) [reply h1.c]: Fixed

**Pull request merged**